api/src/test/java/org/openmrs/api/db/hibernate/HibernateConditionDAOTest.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.api.db.hibernate;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Concept;
+import org.openmrs.ConceptName;
+import org.openmrs.Condition;
+import org.openmrs.api.db.ConditionDAO;
+import org.openmrs.ConditionClinicalStatus;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Patient;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateConditionDAOTest extends BaseContextSensitiveTest {
+	
+	private static final String CONDITIONS_XML = "org/openmrs/api/db/hibernate/include/HibernateConditionDAOTestDataSet.xml";
+	
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+	
+	@Autowired
+	ConditionDAO dao;
+	
+	
+	@Before
+	public void setUp() {
+		executeDataSet(CONDITIONS_XML);
+		
+		updateSearchIndex();
+	}
+	
+	@Test
+	public void shouldSaveCondition() {
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText(new Concept(4), new ConceptName(5089), "non coded");
+		ConditionClinicalStatus clinicalStatus = ConditionClinicalStatus.ACTIVE;
+		ConditionVerificationStatus verificationStaus = ConditionVerificationStatus.CONFIRMED;
+		Patient patient = new Patient(2);
+		Date onsetDate = new Date();
+		Date endDate = new Date();Condition previousVersion = dao.getConditionByUuid("2cc6880e-2c46-15e4-9038-a6c5e4d22fb7");
+		
+		Condition condition = new Condition();
+
+		condition.setConditionId(20);
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(clinicalStatus);
+		condition.setVerificationStatus(verificationStaus);
+		condition.setPreviousVersion(previousVersion);
+		condition.setAdditionalDetail("additionalDetail");
+		condition.setOnsetDate(onsetDate);
+		condition.setEndDate(endDate);
+		condition.setPatient(patient);
+		
+		Condition savedCondition = dao.saveCondition(condition);
+		
```

Comment:
After saving, you need to fetch the condition afresh from the database and then confirm that each property is as you set it.